`solvers`: Implement CLI for 1inch

cowfee opened the pull request:
Progress on https://github.com/cowprotocol/services/issues/882.

Add a CLI command to run the 1inch solver. More or less just boilerplate code.

Follow-up to https://github.com/cowprotocol/services/pull/1250.

### Test Plan

Automated tests.

MartinquaXD commented on crates/solvers/src/infra/config/dex/oneinch/file.rs:
```diff
@@ -0,0 +1,80 @@
+    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
+    endpoint: Option<reqwest::Url>,
+
+    /// The 1Inch liquidity sources to consider when swapping.
+    #[serde(flatten)]
+    liquidity: Liquidity,
```
The `Liquidity` enum is a nice idea but according to the serde [docs](https://serde.rs/container-attrs.html#deny_unknown_fields) `deny_unknown_fields` doesn't play nice with `flatten`.

nlordell commented:
We do have tests. For example, here are the 0x solver tests: https://github.com/cowprotocol/services/blob/main/crates/solvers/src/tests/zeroex/mod.rs

cowfee commented:
Writing the tests.

cowfee commented:
Added tests and fixed a bug that was previously present with expecting `gas` to be a string.

https://github.com/cowprotocol/services/pull/1518/commits/efc52cbbd1375dc908b965945559c0db26cbcbff
